pennylane/fourier/visualize.py

```diff
@@ -98,14 +90,15 @@ def _extract_data_and_labels(coeffs):
         (list(str), dict[str, array[complex]): The set of frequency labels, and a data
             dictionary split into real and imaginary parts.
     """
-    # extract the x ticks
-    nvecs = list(to_dict(coeffs[0]).keys())
+    # extract the x ticks: create generator for indices nvec = (n1, ..., nN),
+    # ranging from (-d1,...,-dN) to (d1,...,dN).
+    nvecs = list(product(*(np.array(range(-(d // 2), d // 2 + 1)) for d in coeffs[0].shape)))
     nvecs_formatted = [format_nvec(nvec) for nvec in nvecs]
 
-    # make data
+    # extract flattened data by real part and imaginary part, and respecting negative indices
     data = {}
-    data["real"] = np.array([[c[nvec].real for nvec in nvecs] for c in coeffs])
-    data["imag"] = np.array([[c[nvec].imag for nvec in nvecs] for c in coeffs])
+    data["real"] = np.array([[c[nvec] for nvec in nvecs] for c in coeffs.real])
+    data["imag"] = np.array([[c[nvec] for nvec in nvecs] for c in coeffs.imag])
```

Comment:
I might be wrong, but I think you can just do `coeffs.real[nvecs]` and `coeffs.imag[nvecs]` here

Reply:
Unfortunately this does not work because of the leading axis that is not to be indexed as well.
I tried to make it work by moving the axis before and after slicing but didn't get it to work right away.
Given that this is not a highly used part of the codebase I'd leave it for now :)